cypress/e2e/03-add_devices.cy.js

```diff
@@ -1,129 +1,129 @@
 import {
   generateCovidOnlyDevice,
   generateMultiplexDevice,
-  loginHooks,
+  loginHooks, testNumber,
 } from "../support/e2e";
 import { graphqlURL } from "../utils/request-utils";
 import { aliasGraphqlOperations } from "../utils/graphql-test-utils";
-const covidOnlyDevice = generateCovidOnlyDevice();
-const multiplexDevice = generateMultiplexDevice();
+import {
+  accessOrganizationByName,
+  cleanUpPreviousRunSetupData,
+  cleanUpRunOktaOrgs,
+  setupRunData
+} from "../utils/setup-utils";
+
+const specRunName = "spec03";
+const currentSpecRunVersionName = `${testNumber()}-cypress-${specRunName}`;
 
-describe("Adding covid only and multiplex devices", () => {
-  let facility;
+describe("Adding testing devices", () => {
+  const covidOnlyDevice = generateCovidOnlyDevice();
+  const multiplexDevice = generateMultiplexDevice();
   loginHooks();
 
-  before("store patient info", () => {
-    cy.task("setCovidOnlyDeviceName", covidOnlyDevice.name);
-    cy.task("setMultiplexDeviceName", multiplexDevice.name);
+  before("setup spec data", () => {
+    cy.task("getSpecRunVersionName", specRunName)
+      .then((prevSpecRunVersionName) => {
+        if (prevSpecRunVersionName) {
+          cleanUpPreviousRunSetupData(prevSpecRunVersionName);
+          cleanUpRunOktaOrgs(prevSpecRunVersionName);
+        }
+        let data = {
+          specRunName: specRunName,
+          versionName: currentSpecRunVersionName
+        };
+        cy.task("setSpecRunVersionName", data)
+        setupRunData(currentSpecRunVersionName);
+      })
   });
 
-  context("Add devices", () => {
-    beforeEach(() => {
-      cy.intercept("POST", graphqlURL, (req) => {
-        aliasGraphqlOperations(req);
-      });
+  beforeEach("alias graphql operations", () => {
+    cy.intercept("POST", graphqlURL, (req) => {
+      aliasGraphqlOperations(req);
     });
+  })
```

Comment:
I think we may need to add the cleanup functions to delete the org info from Okta / locally in an after hook. [Spec 9 code for reference](https://github.com/CDCgov/prime-simplereport/blob/b85d7978b6282cd1fc59d78c5c272ef4f03b1c69/cypress/e2e/09-multiplex_testing.cy.js#L64)

Reply:
ooo nice catch -- will add that now! 🎣